src/class-parsely.php

```diff
@@ -332,214 +331,21 @@ public static function post_has_trackable_status( $post ): bool {
-			$parsely_page['url']              = $this->get_current_url( 'post', $post->ID );
-			$parsely_page['thumbnailUrl']     = $image_url;
-			$parsely_page['image']            = array(
-				'@type' => 'ImageObject',
-				'url'   => $image_url,
-			);
-
-			$this->set_metadata_post_times( $parsely_page, $post );
-
-			$parsely_page['articleSection'] = $category;
-			$author_objects                 = array();
-			foreach ( $authors as $author ) {
-				$author_tag       = array(
-					'@type' => 'Person',
-					'name'  => $author,
-				);
-				$author_objects[] = $author_tag;
-			}
-			$parsely_page['author']    = $author_objects;
-			$parsely_page['creator']   = $authors;
-			$parsely_page['publisher'] = array(
-				'@type' => 'Organization',
-				'name'  => get_bloginfo( 'name' ),
-				'logo'  => $parsely_options['logo'],
-			);
-			$parsely_page['keywords']  = $tags;
-		} elseif ( in_array( get_post_type(), $parsely_options['track_page_types'], true ) && self::post_has_trackable_status( $post ) ) {
-			$parsely_page['headline'] = $this->get_clean_parsely_page_value( get_the_title( $post ) );
-			$parsely_page['url']      = $this->get_current_url( 'post' );
-		} elseif ( 'page' === get_option( 'show_on_front' ) && ! get_option( 'page_on_front' ) ) {
-			$parsely_page['headline'] = $this->get_clean_parsely_page_value( get_bloginfo( 'name', 'raw' ) );
-			$parsely_page['url']      = home_url();
-		}
-
-		/**
-		 * Filters the structured metadata.
-		 *
-		 * @since 2.5.0
-		 *
-		 * @param array   $parsely_page    Existing structured metadata for a page.
-		 * @param WP_Post $post            Post object.
-		 * @param array   $parsely_options The Parsely options.
-		 */
-		$filtered = apply_filters( 'wp_parsely_metadata', $parsely_page, $post, $parsely_options );
-		if ( is_array( $filtered ) ) {
-			return $filtered;
-		}
-		return array();
-	}
-
-	/**
-	 * Sets all metadata values related to post time.
-	 *
-	 * @since 3.0.2
-	 *
-	 * @param array   $metadata Array containing all metadata. It will be potentially mutated to add keys: dateCreated, dateModified, & datePublished.
-	 * @param WP_Post $post     Post object from which to extract time data.
-	 * @return void
-	 */
-	private function set_metadata_post_times( array &$metadata, WP_Post $post ): void {
-		$date_format      = 'Y-m-d\TH:i:s\Z';
-		$post_created_gmt = get_post_time( $date_format, true, $post );
-
-		if ( false === $post_created_gmt ) {
-			return;
-		}
-
-		$metadata['dateCreated']   = $post_created_gmt;
-		$metadata['datePublished'] = $post_created_gmt;
-		$metadata['dateModified']  = $post_created_gmt;
-
-		$post_modified_gmt = get_post_modified_time( $date_format, true, $post );
-
-		if ( false !== $post_modified_gmt && $post_modified_gmt > $post_created_gmt ) {
-			$metadata['dateModified'] = $post_modified_gmt;
-		}
+		_deprecated_function( 'Parsely::construct_parsely_metadata', '3.3.0' );
```

Comment:
I think we could soft deprecate (just note in the changelog, documentation and DocBlocks) this in 3.3, and then hard deprecate this (call `_deprecated_function()`) in 3.4. It's a smoother experience for the customer developers, and we can't remove this until 4.0 anyway, so there's little point in rushing for us, if internally it's already using the new code anyway.